### ISSUE TITLE
[Map] Feature panel doesn't show when clicking on a marker when zoomed out

### ISSUE THREAD

**Issue by gino-m** [google/ground-platform#587]
1. Open the app without zooming in
2. Click on a marker
3. The map is zoomed and centered on the marker and the URL is updated, but the feature panel appears blank.

The problem occurs intermittently; could there be a race condition between the router and click handler?

![image](https://user-images.githubusercontent.com/228050/103313892-68407c00-49ef-11eb-8efb-32193aee1f0c.png)

**Comment by gino-m** [comment]
@DaoyuT could you PTAL?

**Comment by DaoyuT** [comment]
I updated #554 to solve this. The routing is becoming more and more messy as the map component grows. Sorry I couldn't split it into a dedicated PR for this, it was a tightly coupled with that PR. But documented what I fixed about the router in the PR description. Below is a copy of the description:

Sneaked in some fixes to routing in map component, and here are the details:

 - Subscribe to `selectedFeatureId` and enlarge the marker after drawing all the markers. Previously, routing from `#f=f1` to `#f=f2` won't reduce f1 marker and enlarge f2 marker.
 - Move `panAndZoom` into `selectMarker` function. Previously, routing from `#f=f1` to `#f=f2` won't zoom and focus onto the f2 marker.
 - Enable `setFragmentParam` to update multiple params by accepting a map of keys & values. Previously, if we have `#f=f1&o=o1` and do `setFragmentParam('o', null)`, then `setFragmentParam('f','f2')`, the 2nd operation will actually overwrite the 1st one, resulting in `#f=f2&o=o1`, not `#f=f2` as I would expect. I think the reason is that `getFragmentParams()` is using snapshot of the `activatedRoute`, so we have to set multiple fragment params in one go.

**Comment by gino-m** [comment]
> I updated #554 to solve this. I tried locally, it looks fine. Can you verify once #554 is merged?
> 
> The routing is becoming more and more messy as the map component grows. Sorry I couldn't split it into a dedicated PR for this, it was a tightly coupled with that PR. But documented what I fixed about the router in the PR description. Below is a copy of the description:
> 
> Sneaked in some fixes to routing in map component, and here are the details:
> 
> * Subscribe to `selectedFeatureId` and enlarge the marker after drawing all the markers. Previously, routing from `#f=f1` to `#f=f2` won't reduce f1 marker and enlarge f2 marker.
> * Move `panAndZoom` into `selectMarker` function. Previously, routing from `#f=f1` to `#f=f2` won't zoom and focus onto the f2 marker.
> * Enable `setFragmentParam` to update multiple params by accepting a map of keys & values. Previously, if we have `#f=f1&o=o1` and do `setFragmentParam('o', null)`, then `setFragmentParam('f','f2')`, the 2nd operation will actually overwrite the 1st one, resulting in `#f=f2&o=o1`, not `#f=f2` as I would expect. I think the reason is that `getFragmentParams()` is using snapshot of the `activatedRoute`, so we have to set multiple fragment params in one go.

Thanks, Daoyu, I'll follow up on #554 now.
